src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,537 @@
+﻿using System;
+using System.Drawing;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Globalization;
+
+namespace Iot.Device.CharacterLcd
+{
+    /// <summary>
+    /// This is a high-level interface to an LCD display. 
+    /// It supports automatic wrapping of text, automatic scrolling and code page mappings.
+    /// This class is thread safe, however using Write from different threads may lead to unexpected results, since the order is not guaranteed. 
+    /// </summary>
+    public sealed class LcdConsole : IDisposable
+    {
+        private ICharacterLcd _lcd;
+
+        /// <summary>
+        /// The text currently on the display (required for arbitrary scrolling)
+        /// </summary>
+        private StringBuilder[] _currentData;
+        private LineFeedMode _lineFeedMode;
+        private readonly object _lock;
+        private readonly bool _shouldDispose;
+        private TimeSpan _newLineSleepTime;
+        private string _romType;
+        private Encoding _characterEncoding;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="LcdConsole"/> class using the specified LCD low-level interface. 
+        /// This class automatically configures the low-level interface. Do not use the low-level interface at the same time. 
+        /// </summary>
+        /// <param name="lcd">The low-level LCD interface.</param>
+        /// <param name="romType">Name of character ROM of display. Currently supported types: A00 and A02.</param>
+        /// <param name="shouldDispose">If the class should dispose the LCD driver when it is disposed. Defaults to true</param>
+        public LcdConsole(ICharacterLcd lcd, String romType, bool shouldDispose = true)
+        {
+            _lcd = lcd;
+            _shouldDispose = shouldDispose;
+            _romType = romType;
+            _characterEncoding = null;
+            Size = lcd.Size;
+            _currentData = new StringBuilder[Size.Height];
+            CursorLeft = 0;
+            CursorTop = 0;
+            NewLineSleepTime = TimeSpan.Zero;
+            _lock = new object();
+            _lcd.UnderlineCursorVisible = false;
+            _lcd.BlinkingCursorVisible = false;
+            _lcd.DisplayOn = true;
+            _lcd.BacklightOn = true;
+            ClearStringBuffer();
+            _lcd.Clear();
+        }
+
+        /// <summary>
+        /// Position of the cursor, from left.
+        /// Note: May be outside the bounds of the display.
+        /// </summary>
+        public int CursorLeft
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Position of the cursor, from top
+        /// Note: May be outside the bounds of the display.
+        /// </summary>
+        public int CursorTop
```

Comment:
I think this should be `Point Cursor { get; set; }` (with implementation of set)

Reply:
Also assuming you make it settable please clarify what happens when you send a long text with ScrollUp option and set cursor while it's still displaying

Reply:
Another alternative is match the APIs to be the same as Console (SetCursorPosition etc)

Reply:
Hmm you already have that so this might be just getter in that case (sorry for messy comments, I comment as I review and might be missing context before I read the whole thing)